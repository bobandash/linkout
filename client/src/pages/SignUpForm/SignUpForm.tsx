import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGoogle, faLinkedin } from '@fortawesome/free-brands-svg-icons';

const SignUpForm = () => {
  return (
    <div className="w-full overflow-hidden bg-white p-8 md:w-1/2">
      <h1 className="text-outline text-center font-fingerPaint text-6xl text-white md:hidden">
        Linkout
      </h1>
      <h1 className="text-outline hidden text-center font-fingerPaint text-6xl text-white md:block">
        Sign Up
      </h1>
      <form className="my-5 flex flex-col">
        <label htmlFor="email" className="text-lg">
          Email:
        </label>
        <input
          type="email"
          name="email"
          id="email"
          className="rounded-md bg-gray pl-2 pr-2 text-2xl"
          autoComplete="off"
        />
        <label htmlFor="password" className="mt-4 text-lg">
          Password:
        </label>
        <input
          type="password"
          name="password"
          id="password"
          className="rounded-md bg-gray pl-2 pr-2 text-2xl"
        />
        <label htmlFor="confirmPassword" className="mt-4 text-lg">
          Confirm Password:
        </label>
        <input
          type="password"
          name="confirmPassword"
          id="confirmPassword"
          className="rounded-md bg-gray pl-2 pr-2 text-2xl"
        />
        <button
          type="submit"
          className="mt-5 w-full rounded-md bg-primary p-2 text-3xl font-bold uppercase text-white"
        >
          Sign Up
        </button>
      </form>
      <div className="relative">
        <div className="border-t-1 absolute top-1/2 w-full border border-black"></div>
        <p className="relative z-10 mx-auto w-fit bg-white p-2 text-xl font-bold">
          or sign up with
        </p>
      </div>
      <div className="mb-2 flex flex-col">
        <button className="mb-2 w-full rounded-md bg-secondary p-2 text-lg font-bold uppercase text-white">
          <FontAwesomeIcon icon={faGoogle} /> Google
        </button>
        <button className="w-full rounded-md bg-secondary p-2 text-lg font-bold uppercase text-white">
          <FontAwesomeIcon icon={faLinkedin} /> Linkedin
        </button>
      </div>
      <p>
        Have an account?{' '}
        <Link className="text-blue-400" to={'/'}>
          Sign In
        </Link>
      </p>
    </div>
  );
};

export default SignUpForm;
