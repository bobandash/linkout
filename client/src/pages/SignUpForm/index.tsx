import LinkOutSlogan from './LinkOutSlogan';
import SignUpForm from './SignUpForm';

const index = () => {
  return (
    <div className="relative min-h-screen min-w-full bg-primary">
      <div className="absolute left-1/2 top-1/2 w-10/12 max-w-6xl  -translate-x-1/2 -translate-y-1/2 ">
        <div className="relative flex justify-center rounded-lg shadow-custom">
          <LinkOutSlogan />
          <SignUpForm />
        </div>
      </div>
    </div>
  );
};

export default index;
