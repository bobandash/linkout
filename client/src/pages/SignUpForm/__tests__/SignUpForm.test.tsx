import { render, /* screen */ } from '@testing-library/react';
import SignUpForm from '../index.tsx';
import { describe, it, expect } from 'vitest';
import { MemoryRouter } from 'react-router';
/* import userEvent from '@testing-library/user-event'; */

describe('Sign Up Form Component', () => {
  it('renders form with username, password, and confirm password fields', () => {
    const { container } = render(
      <MemoryRouter>
        <SignUpForm />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  /*   it('correct form data calls api and creates account', () => {
    const user = userEvent.setup();
    const email = screen.getByRole('input', { name: 'username' });
    const password = screen.getByRole('input', { name: 'password' });
    const confirmPassword = screen.getByRole('input', {
      name: 'confirmPassword',
    });
    const submitButton = screen.getByRole('button', { name: 'Sign Up' });

    user.type(email, 'test@gmail.com');
    user.type(password, 'vNyBayVtgXpK!');
    user.type(confirmPassword, 'vNyBayVtgXpK!');
  }); */
});
