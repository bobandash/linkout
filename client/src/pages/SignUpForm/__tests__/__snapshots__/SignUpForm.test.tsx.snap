// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sign Up Form Component > renders form with username, password, and confirm password fields 1`] = `
<div>
  <div
    class="relative min-h-screen min-w-full bg-primary"
  >
    <div
      class="absolute left-1/2 top-1/2 w-10/12 max-w-6xl  -translate-x-1/2 -translate-y-1/2 "
    >
      <div
        class="shadow-custom relative flex justify-center rounded-lg"
      >
        <div
          class="hidden w-1/2 flex-col justify-center overflow-hidden bg-black p-8 text-white md:flex lg:px-12"
        >
          <h1
            class="font-fingerPaint text-6xl uppercase lg:text-7xl xl:text-8xl"
          >
            Linkout
          </h1>
          <p
            class="mt-3 text-2xl"
          >
            Creating more intimate friendships through the inverse
          </p>
        </div>
        <div
          class="w-full overflow-hidden bg-white p-8 md:w-1/2"
        >
          <h1
            class="text-outline text-center font-fingerPaint text-6xl text-white md:hidden"
          >
            Linkout
          </h1>
          <h1
            class="text-outline hidden text-center font-fingerPaint text-6xl text-white md:block"
          >
            Sign Up
          </h1>
          <form
            class="my-5 flex flex-col"
            novalidate=""
          >
            <label
              class="text-lg"
              for="email"
            >
              Email: 
              <span
                class="text-required"
              >
                *
              </span>
            </label>
            <input
              autocomplete="off"
              class="rounded-md bg-gray pl-2 pr-2 text-2xl"
              id="email"
              name="email"
              required=""
              type="email"
            />
            <p
              class="text-error text-sm"
            />
            <p
              class="text-error text-sm"
            />
            <label
              class="mt-4 text-lg"
              for="displayName"
            >
              Display Name:
            </label>
            <input
              class="rounded-md bg-gray pl-2 pr-2 text-2xl"
              id="displayName"
              name="displayName"
              type="text"
            />
            <label
              class="mt-4 text-lg"
              for="password"
            >
              Password: 
              <span
                class="text-required"
              >
                *
              </span>
            </label>
            <input
              class="rounded-md bg-gray pl-2 pr-2 text-2xl"
              id="password"
              name="password"
              required=""
              type="password"
            />
            <p
              class="text-error text-sm"
            />
            <label
              class="mt-4 text-lg"
              for="confirmPassword"
            >
              Confirm Password: 
              <span
                class="text-required"
              >
                *
              </span>
            </label>
            <input
              class="rounded-md bg-gray pl-2 pr-2 text-2xl"
              id="confirmPassword"
              name="confirmPassword"
              required=""
              type="password"
            />
            <p
              class="text-error text-sm"
            />
            <button
              class="mt-5 w-full rounded-md bg-primary p-2 text-3xl font-bold uppercase text-white"
              type="submit"
            >
              Sign Up
            </button>
          </form>
          <div
            class="relative"
          >
            <div
              class="border-t-1 absolute top-1/2 w-full border border-black"
            />
            <p
              class="relative z-10 mx-auto w-fit bg-white p-2 text-xl font-bold"
            >
              or sign up with
            </p>
          </div>
          <div
            class="mb-2 flex flex-col"
          >
            <button
              class="mb-2 w-full rounded-md bg-secondary p-2 text-lg font-bold uppercase text-white"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-google "
                data-icon="google"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 488 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
                  fill="currentColor"
                />
              </svg>
               Google
            </button>
            <button
              class="w-full rounded-md bg-secondary p-2 text-lg font-bold uppercase text-white"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-linkedin "
                data-icon="linkedin"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  fill="currentColor"
                />
              </svg>
               Linkedin
            </button>
          </div>
          <p>
            Have an account?
             
            <a
              class="text-blue-400"
              href="/"
            >
              Sign In
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
